### ISSUE TITLE
Exception thrown if authenticated principal does not have username property

### ISSUE THREAD

**Issue by mahileeb** [grails-plugins/grails-spring-security-core#363]
Using 2.0-RC5 - SpringSecurityGrailsPlugin:787

```
String usernamePropertyName = SpringSecurityUtils.securityConfig.userLookup.usernamePropertyName
User.findWhere((usernamePropertyName): SCH.context.authentication.principal.username)
```

```SCH.context.authentication.principal.username``` should instead be principal.toString (as per doco on UsernamePasswordAuthenticationToken) or use the userLookup.usernamePropertyName config

**Comment by burtbeckwith** [comment]
Why wouldn't it have a username?

**Comment by mahileeb** [comment]
My app has to use an existing hibernate domain model where the user object has an email address and password but no username.

**Comment by mahileeb** [comment]
So we have a custom AuthenticationProvider that returns our own Authentication containing our user object

**Comment by burtbeckwith** [comment]
Would it work if you used the email property as the usernamePropertyName?

**Comment by mahileeb** [comment]
No, SCH.context.authentication.principal.username still fails as there is no username property on the principal object

**Comment by burtbeckwith** [comment]
Ok, thanks for the report, I'll get a fix in for 2.0 final.

**Comment by nayankWebonise** [comment]
@burtbeckwith  : when 2.0 is expected, as i am facing this issue too, also Customer url mapping issue.
